### ISSUE TITLE
Rego code produces unused var warnings in OPA 0.50+

### ISSUE THREAD

**Issue by schrolla** [cisagov/ScubaGear#195]
# 🐛 Summary #

The SCuBAGear repo includes automated checks to validate that Rego policies are well formed accordingly to the OPA engine's strict mode checking. These checks are performed on all branches when a push to Github is performed. As a result, if this action fails, current merge to main rules prevent a merge from happening.

OPA released a new version, 0.50.0, last Friday that included updates to the strict mode checking for unused variables. Specifically, for variables defined in function parameters but not used in those functions. A number of our reporting functions across the various provider's rego policies have versions that define function parameters but don't use them. As a result, the OPA strict mode check with the newest and future versions will fail.

A workaround is currently in place to limit testing to OPA <0.50.0 to prevent testing pipeline failures, but this does not address the root issue long term.

The long-term solution is to address all the 0.50 strict warnings by replacing unused function parameters with the wildcard variable ('_') in each place where it is currently causing an error and test the updated code to ensure it still performs as expected.

## To reproduce ##

1. Download OPA tool v0.50.0+ for Windows (https://openpolicyagent.org/downloads/v0.50.0/opa_windows_amd64.exe)
2. Clone a copy of the ScubaGear repository to the local Windows system
3. Run an OPA strict mode check on the current ScubaGear code base as follows:
    ```
    opa_windows_amd64.exe check Rego Testing/Unit/Rego --strict
    ```
 4. Review the output for any occurrences of unused variable warnings.

## Expected behavior ##

The ScubaGear code should pass an OPA strict mode check without any errors.

## Any helpful log output or screenshots ##
An example of an OPA strict mode check failing in several providers.  These errors should be resolved through code updates.
![Screenshot 2023-03-13 at 10 08 37 AM](https://user-images.githubusercontent.com/108814318/225085431-ab384bc1-f8b9-4f4c-9527-40644c6ec69a.png)

**Comment by mitchel-baker** [comment]
ScubaGear passes OPA strict mode check for the following versions: 

- [ ] v0.50.2
- [ ] v0.51.0
- [ ] v0.52.0
- [ ] v0.53.1
- [ ] v0.54.0
- [ ] v0.55.0
- [ ] v0.56.0
- [ ] v0.57.1

**Comment by mitchelbaker-cisa** [comment]
Resolved in #631
